tests/generation/test_utils.py

```diff
@@ -2048,14 +2063,21 @@ def test_inherits_generation_mixin(self):
         for model_class in self.all_generative_model_classes:
             self.assertTrue("GenerationMixin" in str(model_class.__bases__))
 
-    @require_torch_sdpa
-    @slow
-    def test_eager_matches_sdpa_generate(self):
```

Comment:
The test to check that SDPA = eager is the same as the test to check that FA2 = eager. However, parameterized doesn't work here: we want different `pytest.mark` decorators for each test. As such, the original test was moved to a helper function.